conans/client/build/cmake_toolchain_build_helper.py

```diff
@@ -0,0 +1,142 @@
+import os
+import platform
+
+from conans.client import tools
+from conans.client.build import defs_to_string, join_arguments
+from conans.client.build.cmake_flags import is_multi_configuration, get_generator
+from conans.client.toolchain.cmake import CMakeToolchain
+from conans.client.tools.files import chdir
+from conans.client.tools.oss import cpu_count, args_to_string
+from conans.errors import ConanException
+from conans.model.version import Version
+from conans.util.files import mkdir
+
+
+def _validate_recipe(conanfile):
+    forbidden_generators = ["cmake", "cmake_multi", "cmake_paths"]
+    if any(it in conanfile.generators for it in forbidden_generators):
+        raise ConanException("Usage of toolchain is only supported with 'cmake_find_package'"
+                             " or 'cmake_find_package_multi' generators")
+
+
+def _compute_build_flags(conanfile, generator, parallel, msbuild_verbosity):
+    args = []
+    compiler_version = conanfile.settings.get_safe("compiler.version")
+    if generator and parallel:
+        if ("Makefiles" in generator or "Ninja" in generator) and "NMake" not in generator:
+            args.append("-j%i" % cpu_count(conanfile.output))
+        elif "Visual Studio" in generator and compiler_version and Version(compiler_version) >= "10":
+            # Parallel for building projects in the solution
+            args.append("/m:%i" % cpu_count(output=conanfile.output))
+
+    if generator and msbuild_verbosity:
+        if "Visual Studio" in generator and compiler_version and Version(compiler_version) >= "10":
+            args.append("/verbosity:%s" % msbuild_verbosity)
+
+    return args
+
+
+class CMakeToolchainBuildHelper(object):
+    """ CMake helper to use together with the toolchain feature, it has the same interface
+        as the original 'conans.client.build.cmake.CMake' helper, but it will warn the
+        user about arguments forbidden, not used,... and how to achieve the same behavior
+        with the toolchain.
```

Comment:
This comment is no longer true

```suggestion
    """ CMake helper to use together with the toolchain feature, it has the same interface
        as the original 'conans.client.build.cmake.CMake' helper, but accept only meaninful
        arguments for this new build paradigm.
```

Reply:
Right, I am writing a new docstring for this one.